platform-api/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/server/SshService.java

```diff
@@ -0,0 +1,187 @@
+        String name = null;
+        String privateKey = null;
+        String publicKey = null;
+
+        while (formData.hasNext()) {
+            FileItem item = formData.next();
+            String fieldName = item.getFieldName();
+            switch (fieldName) {
+                case "service":
+                    service = item.getString();
+                    break;
+                case "name":
+                    name = item.getString();
+                    break;
+                case "privateKey":
+                    privateKey = item.getString();
+                    break;
+                case "publicKey":
+                    publicKey = item.getString();
+                    break;
+            }
+        }
+
+        requiredNotNull(service, "Service name required");
+        requiredNotNull(name, "Name required");
+        if (privateKey == null && publicKey == null) {
+            throw new BadRequestException("");//TODO Think about message.
+        }
+
+        sshManager.createPair(getCurrentUserId(), new SshPairImpl(service, name, publicKey, privateKey));
+
+        // We should send 200 response code and body with empty line
+        // through specific of html form that doesn't invoke complete submit handler
+        return Response.ok("", MediaType.TEXT_HTML).build();
+    }
+
+    @POST
+    @Consumes({MediaType.APPLICATION_JSON})
+    public void addPair(SshPairDTO sshPair) throws BadRequestException {
+        requiredNotNull(sshPair, "Ssh pair required");
+        requiredNotNull(sshPair.getService(), "Service name required");
+        requiredNotNull(sshPair.getName(), "Name required");
+        if (sshPair.getPublicKey() == null && sshPair.getPrivateKey() == null) {
+            throw new BadRequestException("");//TODO Think about message.
+        }
+
+        sshManager.createPair(getCurrentUserId(), sshPair);
+    }
+
+    /**
+     * Get the ssh pair by the name of pair and name of service owned by the current user
+     */
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("{service}/{name}")
+    public SshPairDTO getPair(@PathParam("service") String serviceName,
+                              @PathParam("name") String name) throws NotFoundException {
+        return asDto(sshManager.getPair(getCurrentUserId(), serviceName, name));
+    }
+
+    @DELETE
+    @Path("{service}/{name}")
+    public void removePair(@PathParam("service") String serviceName,
+                           @PathParam("name") String name) {
+        sshManager.removePair(getCurrentUserId(), serviceName, name);
+    }
+
+    @GET
+    @Path("{service}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public List<SshPairDTO> getKeys(@PathParam("service") String serviceName) {
+        return sshManager.getPairs(getCurrentUserId(), serviceName)
+                         .stream()
+                         .map(SshService::asDto)
+                         .collect(Collectors.toList());
+    }
```

Comment:
You are using different order of annotations. It makes reading code harder. Please, reorganize them